src/d/actor/d_a_player_swim.inc

```diff
@@ -390,65 +616,107 @@ BOOL daPy_lk_c::procSwimUp() {
 }
 
 /* 8013DB24-8013DCDC       .text procSwimWait_init__9daPy_lk_cFi */
+/*
+ * Initializes the Swim Wait state, where Link treads water while idle.
+ *
+ * Sets up the swim animation with appropriate playback speed, vertical position, 
+ * and initializes the water interaction flags. Disables gravity during this process. 
+ * Based on the parameter, either plays the splash sound and effects or keeps the character idle.
+ * 
+ * Parameters:
+ *   - param_1: If TRUE, adjust Link's velocity based on his progress in the swimming animation
+ * 
+ * Returns:
+ *   - TRUE when initialization of the swim wait process is successful.
+ */
 BOOL daPy_lk_c::procSwimWait_init(BOOL param_1) {
-    float fVar2;
+    float swim_animation_state;
     
     J3DFrameCtrl* mFrameCtrl = mFrameCtrlUnder;
 
     commonProcInit(daPyProc_SWIM_WAIT_e);
 
     if (param_1) {
-        fVar2 = mFrameCtrl->getFrame() / (float)mFrameCtrl->getEnd();
-        mVelocity = speedF * (1.0f - daPy_HIO_swim_c0::m.field_0x60) + speedF * std::fabsf(cM_fcos(fVar2 * M_PI)) * daPy_HIO_swim_c0::m.field_0x60;
+        // Adjust Link's velocity based on his forward speed and his current position within his swimming animation
+        /*
+            Given:
+            s: forward speed (`speedF`)
+            x: Link's current animation position (`mFrameCtrl->getFrame()`)
+            swim_animation_state: Represents (as a percentage) how far Link is through one cycle of his swimming animation
```

Comment:
This local var ranges from 0.0-1.0? I feel like "state" isn't the clearest name for it, that sounds like a mode or state machine.

Maybe `anim_blend`/`swim_anim_blend`?